### DISCUSSION TITLE
Generate XML from a XSD type

### DISCUSSION THREAD

**Discussion by t-arn** [sissaschool/xmlschema#406] [Q&A]
Hello folks,
I have following WSDL https://[wsg.isceco.admin.ch/Livestock/AnimalTracingWSDL/3](https://wsg.isceco.admin.ch/Livestock/AnimalTracingWSDL/3) which contains several XML schemas and I would like to generate an XML string for specific XSD types, for example WriteSheepArrivalBatchNotificationV2

The XML to be generated for WriteSheepArrivalBatchNotificationV2 would be the following:
```
<ns1:WriteSheepArrivalBatchNotificationV2 xmlns:ns1="http://www.admin.ch/xmlns/Services/evd/Livestock/AnimalTracing/1" xmlns:ns2="http://schemas.microsoft.com/2003/10/Serialization/Arrays">
    <ns1:WriteSheepArrivalBatchNotificationRequestV2>
        <ns1:RequestMetadata>
            <ns1:ManufacturerKey></ns1:ManufacturerKey>
            <ns1:LCID></ns1:LCID>
            <ns1:AppName></ns1:AppName>
            <ns1:AppVersion></ns1:AppVersion>
            <ns1:OperatingSystem></ns1:OperatingSystem>
        </ns1:RequestMetadata>
        <ns1:TVDNumber></ns1:TVDNumber>
        <ns1:TVDNumberOrigin></ns1:TVDNumberOrigin>
        <ns1:EventDate></ns1:EventDate>
        <ns1:SheepEartagNumbers>
            <ns2:string></ns2:string>
        </ns1:SheepEartagNumbers>
    </ns1:WriteSheepArrivalBatchNotificationRequestV2>
</ns1:WriteSheepArrivalBatchNotificationV2>
```
Can the xmlschema library be used for such XML generations?
If yes, could you give me some hints on how to actually do this?
Do I first need to create a single XSD with imports for sub-schemas before I can use the xmlschema library?

Thanks in advance
Tom

**Comment by brunato** [comment]
Hi,

> Can the xmlschema library be used for such XML generations?

I'm sorry but this library doesn't support generation of XML from schemas, because that topic is not strictly related to validation of the model (ModelVisitor class is the core of the model validation), and also because XML data generation is often a matter of choices and particular needs, that I don't know and neither I have similar needs.

> Do I first need to create a single XSD with imports for sub-schemas before I can use the xmlschema library?

No you don't, the xmlschema library has two alternatives:

-  Manual imports/includes (but it could be complex, because often you have to delay the build of the schema)
- Providing a list of XSD sources instead of one

Anyway these are non-standard procedure, the best option when your schemas are partitioned, is to build an XSD for your target namespace that define imports and includes of all other schemas.

**Reply by t-arn** [reply]
Hi,
Thank you for your fast reply!
Best regards
Tom
